### ISSUE TITLE
Ability to annotate multiple interfaces for single provider

### ISSUE THREAD

**Issue by apawelec** [uber-go/fx#811]
Lets say I have two command/query interfaces: `AccountCommandRepository` and `AccountQueryRepository` and one infrastructure component `databaseAccountRepository` that implements them both. 

My current configurations looks like this:
```
fx.Provide(func(...) (command.AccountCommandRepository, query.AccountQueryRepository) {
    r := newDatabaseAccountRepository(...)
    return r, r
}),
```

If I'm not wrong, there is currently no easy way to mimic this with `fx.Annotate`/`fx.As`, the only way I've figured is:

```
fx.Provide(
    fx.Annotate(
        func(...) (*databaseAccountRepository, *databaseAccountRepository) {
            r := newDatabaseAccountRepository(...)
            return r, r
        },
        fx.As(new(command.AccountCommandRepository), new(query.AccountQueryRepository)),
    ),
),
```
but it's even more verbose (of course I can modify `newDatabaseAccountRepository` constructor to return same value twice, to simplify this, but I prefer don't do this).

It would be nice if this can be replaced by simple:

```
fx.Provide(
    fx.Annotate(
        newDatabaseAccountRepository,
        fx.As(new(command.AccountCommandRepository)),
        fx.As(new(query.AccountQueryRepository)),
    ),
),
```
but in current version it's not possible and results with error `cannot apply more than one line of As`.

Could you consider to change this behaviour?

**Comment by abhinav** [comment]
Hello! Yes, supporting that was next on the list for `fx.As`.
The latter version, specifically:

```go
fx.Provide(
    fx.Annotate(
        newDatabaseAccountRepository,
        fx.As(new(command.AccountCommandRepository)),
        fx.As(new(query.AccountQueryRepository)),
    ),
),
```

We'll welcome a PR adding support for that,
but if you don't have time to contribute this,
we plan to work on this in the coming weeks.

Internal issue ref: GO-1023

**Comment by sywhang** [comment]
This has been implemented in #817. Closing this issue as done.
